zebracrossing/campaigns/views.py

```diff
@@ -9,17 +9,43 @@
 from django.views.generic.edit import CreateView
 from django.urls import reverse
 
-from .models import BookingSheet, Campaign, Material
+from .models import BookingSheet, Campaign, Material, BookedDay
 from .forms import BookingSheetForm, CampaignForm, MaterialForm
+from rest_framework.views import APIView
+from rest_framework.response import Response
 
 
 class CampaignView(mixins.LoginRequiredMixin, DetailView):
     model = Campaign
+    def schedule_data(self, **kwargs):
+        sheet = BookingSheet.objects.filter(campaign=self.get_object())[0]
+        return sheet
 
 
 class BookingView(mixins.LoginRequiredMixin, DetailView):
     model = BookingSheet
 
+class scheduleList(APIView):        
+    def get(self, request):
+        start_date = request.GET.get("start")
+        end_date = request.GET.get("end")
+        my_dict = {}
+        my_array = []
+        if start_date is not None and end_date is not None:
+            schedule = BookedDay.objects.filter(date__range=[start_date, end_date])
+            for item in schedule:
+                if item.bookingsheet.campaign.client not in my_dict:
```

Comment:
This condition isn't required? It will always be false too because it will look for the client name as a key